### ISSUE TITLE
A question about meta-learning few-shot NER

### ISSUE THREAD

**Issue by wjczf123** [microsoft/vert-papers#38]
Hi, I have a question about ablation study in Table 4 1) w/o MAML.

I run the code to get the results of w/o MAML and find that w/o MAML also uses learner.evaluate_meta_() to evalate the performance.

As you know, meta learning aims to fast adaption. It is reasonable for your method to use evaluate_meta_() to fast adaption, but w/o MAML uses evaluate_meta_() to evaluate is unreasonable. 

From my view, w/o MAML also trained the model again during the meta-testing phase (fast adaption). However, this method is w/o MAML.

**Comment by iofu728** [comment]
Hi @wjczf123, thanks for your questions.
First, I want to clarify that the scene is an N-ways K-shot few-shot task, which means it can only use k examples in each test-task support set after using all support/query training sets. So when we do the w/o MAML experiment, there're two ways.
1. Finetuning in support/query training set, and fine-tune in each k example.
2. Finetuning in support/query training set + k test support example for each task.

The second way has high training costs and is prone to catastrophic forgetting. As we mention in our paper, we use the first way to do the w/o MAML ablation experiment.

The second one I want to clarify is that the `evaluate_meta_` function only fine-tunes several steps(both entity span detection and entity typing) in each support test set, and evaluates the performance in each query test set. And the evaluation logits are almost the same between w MAML and w/o MAML. So we reuse them as the same function `evaluate_meta_`. And we also search the hyperparameter by the validation set. According to our experience, there're also convergence fine-tuning steps for each module and each task.

And I agree with your opinion, that MAML has advantages in this scenario. It's also our motivation to explore a way to resolve the few-shot NER problem.

**Comment by wjczf123** [comment]
Thanks for your reply. My core point is that when MAML is not used, there should be no fine-tuning in meta-testing stage.

As you know, w/o MAML methods (i.e., NNShot and ESD) that usually do not use support set and fine-tune in meta-testing stage. These methods only use support set in meta-testing to construct prototype.

Due to MAML methods aim to fast adaption, they fine-tune in meta-testing. So I argue that w/o MAML may be not fine-tune in meta-testing stage. Fine-tuning on support test set will make the result better.

**Comment by wjczf123** [comment]
Can you report this result in this repository?

**Comment by iofu728** [comment]
Hi @wjczf123, I understand your consideration. Sure, as I mentioned before, we search the fine-tune steps hyperparameter(including 0 steps, zero-shot learning) in the w/o MAML experiments. In fact, most of the time, the entity span module needs several fine-tuning steps, especially in Few-NERD Intra(the gap is big, more than 10 points f1-score) and Cross-Domain datasets. And in the entity typing module, most of the time, the optimized hyperparameter is 1 step, even 0 steps(the gap is very small, less than 1 point f1-score).

**Comment by wjczf123** [comment]
OK. Thanks for your detailed reply.
